### DISCUSSION TITLE
The supplied certificate contains no private key. Constructing the certificate with the 'X509KeyStorageFlags.MachineKeySet' flag may solve this problem.

### DISCUSSION THREAD

**Discussion by damonjanis** [ravendb/ravendb#14304] [Q&A] [answered]
Deploying app to Azure App Service. The pfx certificate doesn't load, this error is returned:

The supplied certificate contains no private key. Constructing the certificate with the 'X509KeyStorageFlags.MachineKeySet' flag may solve this problem.

I added X509KeyStorageFlags.MachineKeySet to the certificate creation and then get "Invalid Signature" when the code does SaveChanges.

Any tips on how to configure for Azure App Services?

**Comment by ayende** [comment]
Are you sure that the `PFX` file you are using actually have a private key? The `MachineKeySet` error is typically different.
Can you inspect the certificate locally?

**Comment by damonjanis** [comment]
To inspect locally I ran 

`openssl pkcs12 -in [filename of pfx file].pfx -nocerts -out priv-key.pem -nodes`

The private key was put in the priv-key.pem file so the pfx does contain the private key.

The pfx file is downloaded from my RavenDB Cloud account. It works fine connecting when running on my local machine. However once I deploy to Azure App Services and it runs there it throws this error.

The code that creates the certificate is:

`X509Certificate2 certificate = new X509Certificate2(File.ReadAllBytes("[name of certificate here].pfx"));`

The pfx file is set to Build Action "EmbeddedResource" and Copy To Output Directory is set to "Always Copy".

**Reply by ayende** [reply]
And there, when you call to `X509KeyStorageFlags.MachineKeySet`, ? Also, does `certificate.HasPrivateKey` returns true?

**Comment by damonjanis** [comment]
Yes, when creating certificate with `X509KeyStorageFlags.MachineKeySet` the `certificate.HasPrivateKey` returns `true`

**Comment by garayx** [comment, marked as answer]
Try to set `WEBSITE_LOAD_USER_PROFILE=1` and create the cert with `X509KeyStorageFlags.MachineKeySet`

https://docs.microsoft.com/en-us/azure/app-service/configure-ssl-certificate-in-code#load-certificate-from-file
